guide/how-it-works/ecash/design-best-practices.md

```diff
@@ -0,0 +1,496 @@
+
+It's useful to display generated tokens in the transaction history along with their status—whether pending or redeemed. This helps users quickly assess their transaction status. Providing the ability to expand on a token to view details and take actions like resending or redeeming is also important. This feature becomes particularly valuable if a token was lost or never claimed.
+
+## Cashu best practices
+
+### Default Mint
+{% include image-gallery.html pages = page.images_mint-default %}
+When designing for Cashu, consider allowing users to set a default mint. Since Cashu mints are independently operated, users may prefer to use a trusted mint for their transactions. By allowing them to select a default mint, you simplify their experience and reduce the risk of exposure to a potentially malicious mint. Additionally, consider offering auto-swap preferences.
+
+### Auto-Swapping to a Default Cashu Mint
+{% include image-gallery.html pages = page.images_auto-swap %}
+Wallets can provide the option to automatically swap ecash from an unknown mint to their a default mint. This optional feature can simplify transactions by routing all ecash transactions through the assigned default mint. An auto-swap requires a lightning payment..
+
+{% include tip/open.html color="blue" icon="info" label="Auto-Swapping Involves Fees" %}
+
+It's important to inform users that auto-swapping involves fees and that the amount depends on the amount of ecash they want to swap and the current network conditions.
+
+{% include tip/close.html %}
+
+
+### Backup and restore for cashu
+The Cashu backup and restoration process is designed to ensure users can securely recover their wallets and maintain access to their bitcoin backed ecash tokens in the event they switch devices or experience data loss. **Users should only use their recovery seed phrase once.** Repeated use of the seed phrase for restoration can lead to synchronization issues and potential errors. This is due to the fact that each time you restore, you might be dealing with an outdated state of the wallet, which can cause discrepancies in token balances and transactions.
+
+{% include tip/open.html color="blue" icon="info" label="Single Use Recovery" %}
+
+After successfully restoring your wallet using a recovery seed phrase, it is highly recommended that you generate a new recovery seed phrase immediately. This step is crucial because the original recovery phrase has now been used and could be more susceptible to synchronization risks.
+
+{% include tip/close.html %}
+
+
+### Backup process
+{% include image-gallery.html pages = page.images_backup %}
+
+1. **Deterministic Wallet with Seed Phrase**: Cashu uses a deterministic wallet model, where all cryptographic keys and bitcoin backed ecashtokens can be derived from a single seed phrase. This seed phrase can be generated when the wallet is first created, or a at any point in time when the user wants to backup their wallet.
```

Comment:
The seed can't be generated after the fact as previously generated could not be derived from it. In this case the wallet would have to swap proofs with secrets generated in some other way from proofs with secrets derived from the seed.

Reply:
When you say “after the fact,” do you mean after a user has used the wallet to get ecash? Does this mean the user _must_ generate and back up their seed phrase when the Cashu wallet is created, unlike Bitcoin wallets where you can back up your wallet at any time?

Reply:
Only ecash derived from the seed can be restored later from that seed. So it depends what you mean by back up. A wallet could create a seed when the wallet is created but not show the user until they decide to go back it up later but be using the seed in the background to derive the secrets for the ecash. In this case the user can back the seed up later the still restore as the seed was being used from the start they just didn't know. 

But if a wallet is created without a seed and secrets are derived randomly and then later the user creates a seed and backs it up the randomly derived secrets could not be restored from the seed.  This is also true for bitcoin wallets. 

I think the best ux is for the wallet to generate a seed and prompt the user back it up when the wallet it created but **not** enforce them to back it by reentering the words. And then maybe at a set period or if the wallet balance reaches a certain amount prompt again to backup until the users confirm they did.

Reply:
Agreed. The practice of having the wallet generate a backup at the time of wallet creation is the pattern we should follow. I wasn’t aware this applied to bitcoin wallets as well, where if a seed is created after users start using the wallet, any funds received prior to seed creation would be lost. I'm guessing most bitcoin wallets communicating al this to the user by automatically generating a seed phrase during wallet creation. It’s up to the wallet’s UX to decide whether to present the seed phrase backup process during onboarding or let users handle it later.